Mage/src/main/java/mage/abilities/Ability.java

```diff
@@ -502,6 +502,13 @@ default boolean hasTapCost() {
      */
     UUID getOriginalId();
 
+    /**
+     * Sets a replacement for the name before substituting {this} in rule text.
+     *
+     * @param nameReplacement
+     */
+    Ability withNameReplacement(String nameReplacement);
```

Comment:
Hm, not convinced we should actually add this, though I have thought about similar things and it's possible this is best

Reply:
eh sure, it's good enough